tests/phpunit/tests/option/option.php

```diff
@@ -524,4 +524,147 @@ public function data_update_option_type_juggling() {
 			array( null, array(), true ),
 		);
 	}
+
+	/**
+	 * @ticket 22192
+	 *
+	 * @covers ::update_option
+	 *
+	 * @dataProvider data_update_option_type_juggling
+	 *
+	 * @param mixed $old_value One of the values to compare.
+	 * @param mixed $new_value The other value to compare.
+	 */
+	public function test_update_option_should_hit_cache_when_loosely_equal_to_existing_value_and_cached_values_are_faithful_to_original_type( $old_value, $new_value ) {
+		add_option( 'foo', $old_value );
+		$num_queries = get_num_queries();
+
+		$updated = update_option( 'foo', $new_value );
+
+		$this->assertFalse( $updated, 'update_option should not return true when values are loosely equal.' );
+		$this->assertSame( $num_queries, get_num_queries(), 'The number of database queries should not change.' );
+	}
+
+	/**
+	 * @ticket 22192
+	 *
+	 * @covers ::update_option
+	 *
+	 * @dataProvider data_update_option_type_juggling
+	 *
+	 * @param mixed $old_value One of the values to compare.
+	 * @param mixed $new_value The other value to compare.
+	 */
+	public function test_update_option_should_hit_cache_when_loosely_equal_to_existing_value_and_cached_values_are_pulled_from_the_database( $old_value, $new_value ) {
+		add_option( 'foo', $old_value );
+		wp_cache_delete( 'alloptions', 'options' );
+		wp_load_alloptions();
+
+		$num_queries = get_num_queries();
+
+		$updated = update_option( 'foo', $new_value );
+
+		$this->assertSame( (string) $old_value, get_option( 'foo' ), 'The value should not have been changed.' );
+		$this->assertFalse( $updated, 'update_option should not return true when values are loosely equal.' );
+		$this->assertSame( $num_queries, get_num_queries(), 'The number of database queries should not change.' );
+	}
+
+	/**
+	 * @ticket 22192
+	 *
+	 * @covers ::update_option
+	 *
+	 * @dataProvider data_update_option_values_loosely_equal_to_false
+	 *
+	 * @param mixed $old_value One of the values to compare.
+	 * @param mixed $new_value The other value to compare.
+	 * @param bool  $expected  The expected result.
+	 */
+	public function test_update_option_with_falsey_values( $old_value, $new_value, $expected ) {
```

Comment:
This test and the related dataProvider are probably no longer needed with the inclusion of these use cases in the tests added in https://core.trac.wordpress.org/changeset/56648.